compiler/resolution/lowerIterators.cpp

```diff
@@ -1528,7 +1553,7 @@ getIteratorChildren(Vec<Type*>& children, Type* type) {
 
 static void
 buildIteratorCallInner(BlockStmt* block, Symbol* ret, int fnid, Symbol* iterator) {
-  IteratorInfo* ii = iterator->type->defaultInitializer->getFormal(1)->type->defaultInitializer->iteratorInfo;
+  IteratorInfo* ii = getTheIteratorFn(iterator)->iteratorInfo;
```

Comment:
Alas I do not know what that relationship is and where that documentation lives.
I am leaving that outside of this patch.